### PR TITLE
Ignore caches if GT4Py Version that generated the cache does not match

### PR DESCRIPTION
**Pull request by twicki**
## Purpose

If only the gt4py version changes, our current jenkins infrastructure copies in all the previously generated files and then runs with `rebuild=False`. This has the effect that changes to gt4py are never picked up. This PR changes it such that the gt4py version is written to the cache and the cache is only copied in if the versions match.

## Infrastructure changes:

- All the jenkins plans started from `gt4py-develop` are rebuilding everything
- If the `GT4PY_VERSION` changes, fv3core tests will rebuild everything

### CODE REVIEW

**Comment by rheacangeo** [comment]
you need to 'make lint'

File: .jenkins/actions/run_cached_fv_dynamics.sh
Comment (ofuhrer) [inline .jenkins/actions/run_cached_fv_dynamics.sh:12]: From where is `$GT4PY_VERSION` set here?

Comment (ofuhrer) [inline .jenkins/actions/run_cached_fv_dynamics.sh:12]: Just learned that `=` is equivalent to `==` in bash. This is fine, but I think we mostly use `==` in our scripts.

**Review by ofuhrer — APPROVED** [review]
This looks good to me. Two questions:
- How is "All the jenkins plans started from gt4py-develop are rebuilding everything" implemented?
- Would it make sense to manually add a `version.txt` to the caches right now?

File: .jenkins/actions/run_cached_fv_dynamics.sh
Comment (twicki) [reply .jenkins/actions/run_cached_fv_dynamics.sh:12]: from here:
https://github.com/VulcanClimateModeling/fv3core/blob/415cfb9356a6c9cea697c6cea70989865eb2ca3b/.jenkins/jenkins.sh#L131

Comment (twicki) [reply .jenkins/actions/run_cached_fv_dynamics.sh:12]: bash doing bashy thing. I'm fine following any style recommendations.

**Comment by twicki** [comment]
Some replies to @ofuhrer questions:
> how is "All the jenkins plans started from gt4py-develop are rebuilding everything" implemented?

The plan that launches fv3core regression from a new version of  gt4py sets the variable to the hash (which never matches the version)
Jenkins config

> Would it make sense to manually add a `version.txt` to the caches right now?

Yes it would! I just assumed we are fiddling with the cache now so it might get deleted again until we merge this (which we can only do once we're back to green) - this is why i put it off for now

File: .jenkins/actions/run_cached_fv_dynamics.sh
Comment (ofuhrer) [reply .jenkins/actions/run_cached_fv_dynamics.sh:12]: Maybe comment it?

**Comment by ofuhrer** [comment]
Thanks. All good from me.

**Comment by rheacangeo** [comment]
if we do #335 , we should use that here. otherwise, should change the check for GT4PY_VERSION to be like the other ones in master now

**Comment by rheacangeo** [comment]
These changes are merged into #335, so we can close this one

**Comment by twicki** [comment]
launch jenkins

**Comment by twicki** [comment]
closed in favor of #335

**Pull request closed without merging**
